commons/src/test/scala/dagr/commons/util/StringUtilTest.scala

```diff
@@ -21,36 +21,48 @@
  * THE SOFTWARE.
  */
-
-package dagr.core.util
+package dagr.commons.util
 
 /**
   * Tests for StringUtil
   */
 class StringUtilTest extends UnitSpec {
-  "StringUtil.camelToGnu" should "correctly handle simple multi-word variable names" in {
-    StringUtil.camelToGnu("fooBar") should be ("foo-bar")
-    StringUtil.camelToGnu("oneTwoThreeFourFiveSixSeven") should be ("one-two-three-four-five-six-seven")
-  }
 
-  it should "not modify single word variable names" in {
-    StringUtil.camelToGnu("foo") should be ("foo")
+  "StringUtil.wrapString" should "wrap a string with a prefix and suffix" in {
+    StringUtil.wrapString("pre-", "body", "-suffix") shouldBe "pre-body-suffix"
   }
 
-  it should "replace initial caps but not insert a hyphen at the start of a word" in {
-    StringUtil.camelToGnu("BigBad") should be ("big-bad")
+  "StringUtil.wordWrap" should "wrap an input string with newlines" in {
+    StringUtil.wordWrap("a b bb b c", maxLength=1) shouldBe "a\nb\nbb\nb\nc"
   }
 
-  it should "insert hyphens between adjacent capital letters" in {
-    StringUtil.camelToGnu("TisASillyName") should be ("tis-a-silly-name")
-  }
+  "StringUtil.levenshteinDistance" should "compute the levenshtein distance between two strings" in {
+    StringUtil.levenshteinDistance("abc", "abc", 1, 1, 1, 1) shouldBe 0
+    StringUtil.levenshteinDistance("abc", "def", 0, 0, 0, 0) shouldBe 0
 
-  it should "handle single letter names" in {
-    StringUtil.camelToGnu("A") should be ("a")
-    StringUtil.camelToGnu("b") should be ("b")
+
+    // swap
+    StringUtil.levenshteinDistance("abc", "acb", 1, 0, 0, 0) shouldBe 0 // can be achieved using substitutions at no penalty
+    StringUtil.levenshteinDistance("abc", "acb", 0, 0, 0, 0) shouldBe 0
+    StringUtil.levenshteinDistance("abc", "acb", 1, 10, 10, 10) shouldBe 1
+
+    // substitution
+    StringUtil.levenshteinDistance("abc", "abd", 0, 1, 0, 0) shouldBe 0 // can be with an insertion then deletion using substitutions at no penalty
+    StringUtil.levenshteinDistance("abc", "abd", 0, 0, 0, 0) shouldBe 0
+    StringUtil.levenshteinDistance("abc", "abd", 10, 1, 10, 10) shouldBe 1
+
+    // deletion
+    StringUtil.levenshteinDistance("abc", "ab", 0, 0, 0, 1) shouldBe 1 // deletions cannot be achieved with other operators
+    StringUtil.levenshteinDistance("abc", "ab", 0, 0, 0, 0) shouldBe 0
+    StringUtil.levenshteinDistance("abc", "ab", 10, 10, 10, 1) shouldBe 1
+
+    // insertion
+    StringUtil.levenshteinDistance("abc", "abcd", 0, 0, 1, 0) shouldBe 1 // insertions cannot be achieved with other operators
+    StringUtil.levenshteinDistance("abc", "abcd", 0, 0, 0, 0) shouldBe 0
+    StringUtil.levenshteinDistance("abc", "abcd", 10, 10, 1, 10) shouldBe 1
   }
-  
-  "String.formatElapsedTime" should "format the elapsed time in DD:HH:MM:SS" in {
+
+  "StringUtil.formatElapsedTime" should "format the elapsed time in DD:HH:MM:SS" in {
```

Comment:
Not so much a comment on the test, but could we change the format here to be just hours, minutes and seconds?  I find that as soon as there are four fields I struggle to decide if it's days on the front or millis on the back.